### ISSUE TITLE
Docker plugin fails with NullPointerException when using Google Cloud credentials

### ISSUE THREAD

**Issue by cjbrowne** [jenkinsci/docker-plugin#510]
Stack trace below.  I believe the problem is fixed on master, but on version 0.16.2 (I checked out the tag 'docker-plugin-parent-0.16.2') the JenkinsUtil.java:173 will throw a NPE because the method "makeAuthConfig(Credentials credentials)" will return null unless the credentials are StandardUsernamePasswordCredentials, which the Google Cloud Credentials are not.

Surely this makes the docker plugin incompatible with google cloud docker registries?

```
ERROR: Build step failed with exception
java.lang.NullPointerException
	at com.nirima.jenkins.plugins.docker.utils.JenkinsUtils.makeAuthConfig(JenkinsUtils.java:173)
	at com.nirima.jenkins.plugins.docker.utils.JenkinsUtils.getAuthConfigurations(JenkinsUtils.java:150)
	at com.nirima.jenkins.plugins.docker.builder.DockerBuilderPublisher$Run.buildImage(DockerBuilderPublisher.java:273)
	at com.nirima.jenkins.plugins.docker.builder.DockerBuilderPublisher$Run.run(DockerBuilderPublisher.java:230)
	at com.nirima.jenkins.plugins.docker.builder.DockerBuilderPublisher.perform(DockerBuilderPublisher.java:361)
	at hudson.tasks.BuildStepCompatibilityLayer.perform(BuildStepCompatibilityLayer.java:81)
	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:736)
	at hudson.model.Build$BuildExecution.build(Build.java:206)
	at hudson.model.Build$BuildExecution.doRun(Build.java:163)
	at hudson.model.AbstractBuild$AbstractBuildExecution.run(AbstractBuild.java:496)
	at hudson.model.Run.execute(Run.java:1737)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:97)
	at hudson.model.Executor.run(Executor.java:419)
Build step 'Build / Publish Docker Containers' marked build as failure
```

**Comment by ndeloof** [comment]
please confirm issue still exists with 0.17, then reopen if needed

**Comment by abdulhaseebxflow** [comment]
Same issue is being faced by me.

ERROR: Build step failed with exception
java.lang.NullPointerException
	at com.nirima.jenkins.plugins.docker.builder.DockerBuilderControlOptionProvisionAndStart.executeOnDocker(DockerBuilderControlOptionProvisionAndStart.java:56)
	at com.nirima.jenkins.plugins.docker.builder.DockerBuilderControlOptionProvisionAndStart.execute(DockerBuilderControlOptionProvisionAndStart.java:48)
	at com.nirima.jenkins.plugins.docker.builder.DockerBuilderControl.perform(DockerBuilderControl.java:30)
	at hudson.tasks.BuildStepCompatibilityLayer.perform(BuildStepCompatibilityLayer.java:79)
	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:741)
	at hudson.model.Build$BuildExecution.build(Build.java:206)
	at hudson.model.Build$BuildExecution.doRun(Build.java:163)
	at hudson.model.AbstractBuild$AbstractBuildExecution.run(AbstractBuild.java:504)
	at hudson.model.Run.execute(Run.java:1815)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:97)
	at hudson.model.Executor.run(Executor.java:429)
Build step 'Start/Stop Docker Containers' marked build as failure
Finished: FAILURE

**Comment by pjdarton** [comment]
To stand any chance of progressing this, we'll need a full repo case.
e.g. Personally, I've no experience of using "google cloud credentials" and I know that they're not part of the docker plugin, so without further details, I'm not sure what they are (at the code level).

Lets say we've got an empty _default_ Jenkins server installation (e.g. latest LTS version) to which we've added the (latest version of the) docker-plugin - how do we go from there to the stack trace you've provided?
